test/helpers/kubectl.go

```diff
@@ -153,6 +156,40 @@ var (
 		"devices":                     "", // Override "eth0 eth0\neth0"
 	}
 
+	aksHelmOverrides = map[string]string{
+		"image.repository":          "mcr.microsoft.com/oss/cilium/cilium",
+		"image.tag":                 "1.12.1",
+		"image.digest":              "",
+		"image.useDigest":           "false",
+		"operator.image.repository": "mcr.microsoft.com/oss/cilium/operator",
+		"operator.image.tag":        "1.12.1",
+		"operator.image.suffix":     "",
+		"operator.image.digest":     "",
+		"operator.image.useDigest":  "false",
+
+		"ipam.mode": "delegated-plugin",
+
+		"tunnel":                 "disabled",
+		"endpointRoutes.enabled": "true",
+
+		"extraArgs": "{--local-router-ipv4=169.254.23.0}",
+
+		// # kubenet pod CIDR
+		"k8s.requireIPv4PodCIDR":              "false",
+		"ipv6.enabled":                        "false",
+		"ipv4NativeRoutingCIDR":               "10.241.0.0/16",
```

Comment:
I'm not sure exactly how this works but if this often varies in the cluster depending on how you provisioned it, then maybe this one should be parameterized as well. This is already parameterized for GKE runs (`grep NATIVE_CIDR` in the test/ dir).

Reply:
Regarding this a well. Make the default value as this and pass it as parameter through cmd line ? 
I saw the GKE code, they are fetching it from the ENV Variable. What should be the ideal way to pass the parameterized this ?

Reply:
Also, how will this affect the test cases being run in the PR ? ( having a parametric value in the test cases.)

Reply:
Environment variables are usually an easy way to do this, because you can then just set up your own environment with the relevant fields and it will just work. This works for either CI type environments or your local development machine.

I'm not quite sure I understand your concern around parametric values. For the IPv4 Native Routing CIDR, I would expect that if it is configured correctly, all tests pass and if it's not configured correctly then some particular forwarding path may experience problems because e.g. masquerading is misconfigured.